src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -537,6 +537,32 @@ DWORD WINAPI CProfilerManager::ParseAttachConfigurationThreadProc(
                                  it != settingsMap.end();
                                  ++it)
                             {
+                                // Assume format is:
+                                // MicrosoftInstrumentationEngine_LogLevel_12341234-1234-1234-1234-123412341234
+                                if (wcsncmp(it->first.c_str(), _T("MicrosoftInstrumentationEngine_LogLevel_"), 40) == 0)
```

Comment:
>if (wcsncmp(it->first.c_str(), _T("MicrosoftInstrumentationEngine_LogLevel_"), 40) == 0) [](start = 32, length = 88)

I'm thinking we might want to provide this via a different element/attribute in the XML description. It could easily be just an attribute:

<AddInstrumentationMethod ConfigPath="..." LogLevel="Errors">

or if we want to leave it open for further expansion, add another element child of AddInstrumentationMethod to specify values meant to be consumed by the engine (rather than the method) but describe method configuration:

<AddInstrumentationMethod ConfigPath="...">
  <Configuration>
    <LogLevel>Errors</LogLevel>
    ...
  </Configuration>
  <Settings>
    ...
  </Settings>
</AddInstrumentationMethod>

Reply:
In either of these cases, we don't have to search for patterns of elements, just for exact names, and we don't force IM producers to set additional settings that they'll see during initialization that they don't really need to consume.

---
In reply to: [400363077](https://github.com/microsoft/CLRInstrumentationEngine/pull/231#discussion_r400363077) [](ancestors = 400363077)

Reply:
I'll add the following to the ConfigurationSources schema:

<InstrumentationConfigurationSource>
 <Platforms>
  ...
  </Platforms>
  <Configuration>
    <LogLevel>Errors Dumps</LogLevel>
  </Configuration>
  <Settings>
  ...
  </Settings>
</InstrumentationConfigurationSource>

This would make it an easy mapping to either an attribute on or <Configuration /> section of <AddInstrumentationMethod />

---
In reply to: [400363975](https://github.com/microsoft/CLRInstrumentationEngine/pull/231#discussion_r400363975) [](ancestors = 400363975,400363077)

Reply:
I'll need to get the MethodGuid somehow in order to track the correct loglevel to method. I'm thinking about adding this logic to the attach handler when parsing the configurationsource.

---
In reply to: [405687799](https://github.com/microsoft/CLRInstrumentationEngine/pull/231#discussion_r405687799) [](ancestors = 405687799,400363975,400363077)